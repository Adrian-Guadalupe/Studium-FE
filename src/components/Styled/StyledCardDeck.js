import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { Card } from 'antd';

const StyledCardDeck = ({
  text,
  type = 'inner',
  size = 'default',
  icon,
  loading,
  block,
  hoverable,
  ...props
}) => {
  return (
    <StyledAntdCard
      type={type}
      size={size}
      icon={icon}
      loading={loading && 'loading'}
      block={block && 'block'}
      {...props}
    >
      {text}
    </StyledAntdCard>
  );
};

StyledCardDeck.propTypes = {
  text: PropTypes.string,  
  type: PropTypes.oneOf(['inner']),
  size: PropTypes.oneOf(['default', 'small']),
  icon: PropTypes.string,
  loading: PropTypes.bool,
  block: PropTypes.bool,
  hoverable: PropTypes.bool,
};

const StyledAntdCard = styled(Card)`
  && {
    width: 94px;
    height: 94px;
    left: 24px;
    top: '240px';
    padding: '0 auto';
    border-radius: 17px;
    background-color: ${props => {
      if (props.type === 'primary') {
        return props.theme.mainColor
      } else {
          return props.theme.white
      }
    }};

    &&{
        border: ${props => {
            if(props.border === 'dashed'){
                return '3px dashed rgba(136, 136, 136, 0.75)'
            } else if (props.border === 'solid' ){
                return '3px solid #c4c4c4'
            }
        }}
    }
  }
  `;

export default StyledCardDeck;
