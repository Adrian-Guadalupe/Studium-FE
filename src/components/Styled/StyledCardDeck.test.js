import React from 'react';
import renderer from 'react-test-renderer';

import StyledCardDeck from './StyledCardDeck';



test('StyledCardDeck render correctly', () => {
    const tree = renderer.create(<StyledCardDeck />).toJSON();
    expect(tree).toMatchSnapshot();
});