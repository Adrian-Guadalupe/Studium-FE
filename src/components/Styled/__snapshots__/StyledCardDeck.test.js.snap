// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledCardDeck render correctly 1`] = `
.c0.c0 {
  margin: 8px;
  font-weight: bold;
  width: 94px;
  height: 94px;
  border-radius: 17px;
}

.c0.c0 > .ant-card-body {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 5px 5px;
  height: 100%;
  overflow: hidden;
  text-overflow: ellipsis;
}

<div
  className="ant-card c0 ant-card-bordered ant-card-type-inner"
>
  <div
    className="ant-card-body"
    style={Object {}}
  />
</div>
`;
